src/proj7BittingCerratoCohenEllmer/bantam/lexer/Scanner.java

```diff
@@ -289,132 +269,86 @@ private boolean isCompleteIdentifier(Stack<Character> spellingStack) {
-                        character > 'Z' && character < '_' ||
-                        character > '_' && character < 'a' ||
-                        character > 'z') {
-                    return false;
-                }
+    private Kind getTokenKind(Stack<Character> spellingStack) {
+        char leadingChar = spellingStack.firstElement();
+        if (validSolo.contains(leadingChar)) {
+            switch (leadingChar) {
+                case '(':
+                    return Kind.LPAREN;
+                case ')':
+                    return Kind.RPAREN;
+                case '{':
+                    return Kind.LCURLY;
+                case '}':
+                    return Kind.RCURLY;
+                case ';':
+                    return Kind.SEMICOLON;
+                case '!': // todo: should this be here or do we need a new method?
+                    return Kind.UNARYNOT;
+                case '.':
+                    return Kind.DOT;
+                case ':':
+                    return Kind.COLON;
+                case ',':
+                    return Kind.COMMA;
+                default:
+                    return null;
             }
-            return true;
-        } // doesnt start with letter
-        return false;
-    }
-
-    private boolean isSlash(String spelling) {
-        return spelling.startsWith("/'");
-    }
-
-    private boolean isString(String spelling) {
-        return spelling.startsWith("\"");
+        } else if (isEOF(spellingStack)) {
+            return Kind.EOF;
+        } else if (leadingChar == '/' && spellingStack.size() != 1) {
+            return Kind.COMMENT;
+        } else if (leadingMathChars.contains(leadingChar)) {
+            String tokenString = makeStackString(spellingStack, true);
+            switch (tokenString) {
+                case "+":
+                case "-":
+                    return Kind.PLUSMINUS;
+                case "*":
+                case "/":
+                    return Kind.MULDIV;
+                case "%":
+                case ">":
+                case "<":
+                case "&&":
+                case ">=":
+                case "<=":
+                case "||":
+                    return Kind.BINARYLOGIC; //todo fix modulus
+                case "++":
+                    return Kind.UNARYINCR;
+                case "--":
+                    return Kind.UNARYDECR;
+                default:
+                    return null;
+            }
+        } else if (leadingChar == '=' && spellingStack.size() == 1) {
+            return Kind.ASSIGN;
+        } else if (leadingChar == '=') {
+            return Kind.COMPARE;
+        } else if (Character.isDigit(leadingChar)) {
+            return Kind.INTCONST;
+        } else if (leadingChar == '"') {
+            return Kind.STRCONST;
+        } else {
+            return Kind.IDENTIFIER; // todo: assumes same as check token with respect to else clause.
```

Comment:
Default Token type should be an error and catch if there are illegal characters scanned.